packages/server/src/middleware/builder.ts

```diff
@@ -64,13 +64,18 @@ async function updateAppUpdatedAt(ctx: BBContext) {
   })
 }
 
-export default async function builder(ctx: BBContext, permType: string) {
+export default async function builder(ctx: BBContext) {
   const appId = ctx.appId
   // this only functions within an app context
   if (!appId) {
     return
   }
-  const isBuilderApi = permType === permissions.PermissionType.BUILDER
```

Comment:
I think it's 6 of 1. You could always pass another perm type in and make the function a no-op. Moving the check externally means this will always operate under the assumption of being a builder middleware - applying the strictest checks.